ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java

```diff
@@ -650,7 +652,7 @@ public Node visitSelectSingle(SqlBaseParser.SelectSingleContext context) {
   }
 
   private String replaceDotColon(String input) {
-    return input.replace(".", "_").replace(":", "__");
+    return input.replace(".", "_").replace("->", "__");
```

Comment:
The use of `"->"` throughput the code should be using a constant. As should `"."`